dashboard/test/controllers/dynamic_config_controller_test.rb

```diff
@@ -0,0 +1,76 @@
+require 'test_helper'
+
+class DynamicConfigControllerTest < ActionController::TestCase
+  test 'inaccessible by non-admins' do
+    get :show
+    assert_response :redirect
+    assert_redirected_to :new_user_session
+    get :gatekeeper_show
+    assert_response :redirect
+    assert_redirected_to :new_user_session
+    post :gatekeeper_set, params: {feature: 'test access', where: '{}', value: 'true'}
+    assert_response :redirect
+    assert_redirected_to :new_user_session
+    post :gatekeeper_delete, params: {feature: 'test access', where: '{}'}
+    assert_response :redirect
+    assert_redirected_to :new_user_session
+
+    sign_in(create(:user))
+    get :show
+    assert_response :forbidden
+    get :gatekeeper_show
+    assert_response :forbidden
+    post :gatekeeper_set, params: {feature: 'test access', where: '{}', value: 'true'}
+    assert_response :forbidden
+    post :gatekeeper_delete, params: {feature: 'test access', where: '{}'}
+    assert_response :forbidden
```

Comment:
You can do something similar

```
    sign_in(create(:user))
    non_admin_actions.each do |action|
      get action
      assert_response :forbidden
    end
```


I am not sure the Post and params make a difference for this case at least.